### PR TITLE
fix(destination):  Added/Modified Stats for assistance in debugging for transformerProxy

### PR DESCRIPTION
**Pull request by sanpj2292**
# Description

We have added new stats
1. transformer_proxy.delivery_request
2. transformer_proxy.delivery_response
3. transformer_proxy.req_round_trip_time

For the older stats we didn't have destination as part of tags which was making things difficult while debugging. Added them as well

As part of the [CFD](https://www.notion.so/rudderstacks/Duplicate-events-in-downstream-destinations-RudderStack-ad13926a1153441091aced26bb1b157d?d=f7c7d944be8a415da74c7cb8c950d96c), we're adding more stats to improve the debuggability through this PR

## Notion Ticket

[Duplicate Events in downstream destinations](https://www.notion.so/rudderstacks/Duplicate-events-in-downstream-destinations-RudderStack-ad13926a1153441091aced26bb1b157d?d=f7c7d944be8a415da74c7cb8c950d96c)

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: router/transformer/transformer.go [hunk 1]
```diff
@@ -200,22 +201,19 @@ func (trans *HandleT) ProxyRequest(ctx context.Context, responseData integration
...
-		if requestError != nil {
-			stats.NewTaggedStat("transformer_proxy.request_latency", stats.TimerType, stats.Tags{"requestSuccess": "false"}).SendTiming(time.Since(rdl_time))
-			stats.NewTaggedStat("transformer_proxy.request_result", stats.CountType, stats.Tags{"requestSuccess": "false"}).Increment()
-		} else {
-			stats.NewTaggedStat("transformer_proxy.request_latency", stats.TimerType, stats.Tags{"requestSuccess": "true"}).SendTiming(time.Since(rdl_time))
-			stats.NewTaggedStat("transformer_proxy.request_result", stats.CountType, stats.Tags{"requestSuccess": "true"}).Increment()
-		}
+		respData, respCode, requestError = trans.makeHTTPRequest(ctx, url, payload, destName)
+		stats.NewTaggedStat("transformer_proxy.delivery_response", stats.CountType, stats.Tags{"destination": destName}).Increment()
+		reqSuccessStr := strconv.FormatBool(requestError != nil)
```
Comment (saikumarrs) [inline router/transformer/transformer.go:206]: This check should be inverted to match the `requestSuccess` definition.

File: router/transformer/transformer.go [hunk 2]
```diff
@@ -200,22 +201,19 @@ func (trans *HandleT) ProxyRequest(ctx context.Context, responseData integration
...
+		stats.NewTaggedStat("transformer_proxy.request_result", stats.CountType, stats.Tags{"requestSuccess": reqSuccessStr, "destination": destName}).Increment()
 		//end
 		return requestError
 	}
 
 	backoffWithMaxRetry := backoff.WithMaxRetries(backoff.NewExponentialBackOff(), uint64(retryWithBackoffCount))
 	err = backoff.RetryNotify(operation, backoffWithMaxRetry, func(err error, t time.Duration) {
 		pkgLogger.Errorf("[Transformer Proxy] Request for proxy to URL:: %v, Error:: %+v retrying after:: %v,", url, err, t)
-		stats.NewStat("transformer_proxy.retry_metric", stats.CountType).Increment()
+		stats.NewTaggedStat("transformer_proxy.retry_metric", stats.CountType, stats.Tags{"destination": destName}).Increment()
```
Comment (saikumarrs) [inline router/transformer/transformer.go:216]: Rename it for clarity. 
```suggestion
		stats.NewTaggedStat("transformer_proxy.retry_metric", stats.CountType, stats.Tags{"destination": destName}).Increment()
```

File: router/transformer/transformer.go (hunk 1 above)
Comment (sanpj2292) [reply router/transformer/transformer.go]: Yeah. Changed it

File: router/transformer/transformer.go [hunk 3]
```diff
@@ -200,22 +201,19 @@ func (trans *HandleT) ProxyRequest(ctx context.Context, responseData integration
...
-		respData, respCode, requestError = trans.makeHTTPRequest(ctx, url, payload)
-		if requestError != nil {
-			stats.NewTaggedStat("transformer_proxy.request_latency", stats.TimerType, stats.Tags{"requestSuccess": "false"}).SendTiming(time.Since(rdl_time))
-			stats.NewTaggedStat("transformer_proxy.request_result", stats.CountType, stats.Tags{"requestSuccess": "false"}).Increment()
-		} else {
-			stats.NewTaggedStat("transformer_proxy.request_latency", stats.TimerType, stats.Tags{"requestSuccess": "true"}).SendTiming(time.Since(rdl_time))
-			stats.NewTaggedStat("transformer_proxy.request_result", stats.CountType, stats.Tags{"requestSuccess": "true"}).Increment()
-		}
+		respData, respCode, requestError = trans.makeHTTPRequest(ctx, url, payload, destName)
+		stats.NewTaggedStat("transformer_proxy.delivery_response", stats.CountType, stats.Tags{"destination": destName}).Increment()
```
Comment (saikumarrs) [inline router/transformer/transformer.go:205]: If a request is retried, it'll be difficult to deduce from this value. We can do either of the following:
- change this to represent only success count as `transformer_proxy.retry_metric` is already tracking failures/retries.
- Add the request status as a tag.

File: router/transformer/transformer.go (hunk 2 above)
Comment (saikumarrs) [reply router/transformer/transformer.go:216]: Yes

File: router/transformer/transformer.go (hunk 3 above)
Comment (saikumarrs) [reply router/transformer/transformer.go:205]: Yes.

Comment (sanpj2292) [reply router/transformer/transformer.go]: Changed. Please take a look now

**Review by saikumarrs — APPROVED** [review]

**Review by utsabc — APPROVED** [review]

File: router/router.go [hunk 4]
```diff
@@ -801,6 +804,25 @@ func (worker *workerT) handleWorkerDestinationJobs(ctx context.Context) {
 							}
 						}
 						respBody = strings.Join(respBodyArr, " ")
+						if worker.rt.transformerProxy {
+							stats.NewTaggedStat("transformer_proxy.input_events_count", stats.CountType, stats.Tags{
+								"destType":    worker.rt.destName,
+								"destination": misc.GetTagName(destinationJob.Destination.ID, destinationJob.Destination.Name),
```
Comment (cisse21) [inline router/router.go:810]: let's just use destinationId here

File: router/transformer/transformer.go [hunk 5]
```diff
@@ -198,28 +200,28 @@ func (trans *HandleT) ProxyRequest(ctx context.Context, responseData integration
...
 		//end
 		return requestError
 	}
 
 	backoffWithMaxRetry := backoff.WithMaxRetries(backoff.NewExponentialBackOff(), uint64(retryWithBackoffCount))
 	err = backoff.RetryNotify(operation, backoffWithMaxRetry, func(err error, t time.Duration) {
-		pkgLogger.Errorf("[Transformer Proxy] Request for proxy to URL:: %v, Error:: %+v retrying after:: %v,", url, err, t)
-		stats.NewStat("transformer_proxy.retry_metric", stats.CountType).Increment()
+		pkgLogger.Errorf("[TransformerProxy] (Dest-%[1]v) {Job - %[2]v} Request for proxy to URL:: %[3]v, Error:: %+[4]v retrying after:: %[5]v,", destName, jobId, url, err, t)
+		pkgLogger.Debugf("[TransformerProxy] (Dest-%[1]v) {Job - %[2]v} Request for proxy to URL:: %[3]v, Error:: %+[4]v retrying after:: %[5]v,", destName, jobId, url, err, t)
```
Comment (cisse21) [inline router/transformer/transformer.go:219]: Do we need both error and debug logs here?

**Review by cisse21 — APPROVED** [review]

**Pull request merged**